### PR TITLE
cli: support decimal precision and scale in dump

### LINKED ISSUE #7840: cli: dump command panics when table includes decimal with precision
```
$  grep -F '[config]' cockroach-data/logs/cockroach.INFO
I160714 10:33:43.355096 util/log/clog.go:998  [config] file created at: 2016/07/14 10:33:43
I160714 10:33:43.355096 util/log/clog.go:998  [config] running on machine: JESSEs-MacBook-Pro
I160714 10:33:43.355096 util/log/clog.go:998  [config] binary: CockroachDB beta-20160629-361-gf50771c (darwin amd64, built 2016/07/14 14:28:49, go1.6.2)
I160714 10:33:43.355096 util/log/clog.go:998  [config] arguments: [cockroach start]
```

What I did:

1. I started up a cluster.
2. I opened the interactive sql shell, created a table, and inserted some rows:
   ```
   root@:26257> create test.table dumptest (a int primary key, b decimal(10));
   CREATE TABLE

   root@:26257> insert into test.dumptest values (1, 1.1), (2, 2.2), (3, 3.3);
   INSERT 3
   ```
3. I tried to dump the contents of the new table:
   ```
   $ cockroach dump test dumptest
CREATE TABLE dumptest (
	a INT NOT NULL,
	b DECIMAL(10) NULL,
	CONSTRAINT "primary" PRIMARY KEY (a),
	FAMILY "primary" (a, b)
);
panic: unknown []byte type: 1.1, b: DECIMAL(10)

goroutine 1 [running]:
panic(0x5401380, 0xc82036c440)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
github.com/cockroachdb/cockroach/cli.dumpTable(0x749d330, 0xc820026010, 0xc82033f920, 0x7fff5fbffbc7, 0x4, 0x7fff5fbffbcc, 0x8, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/dump.go:205 +0x429b
github.com/cockroachdb/cockroach/cli.runDump(0x61b72c0, 0xc82033f8e0, 0x2, 0x2, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/dump.go:58 +0x16e
github.com/spf13/cobra.(*Command).execute(0x61b72c0, 0xc82033f720, 0x2, 0x2, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:600 +0x668
github.com/spf13/cobra.(*Command).ExecuteC(0x61b5720, 0x61b72c0, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:690 +0x55c
github.com/spf13/cobra.(*Command).Execute(0x61b5720, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:649 +0x2d
github.com/cockroachdb/cockroach/cli.Run(0xc82000a290, 0x3, 0x3, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/cli.go:95 +0x7c
main.main()
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/main.go:37 +0xe6
   ```

### PR DESCRIPTION
**Pull request by madelynnblue**
Fixes #7840

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7842)

### CODE REVIEW

File: cli/dump.go [hunk 1]
```diff
@@ -192,15 +192,15 @@ func dumpTable(w io.Writer, conn *sqlConn, origDBName, origTableName string) err
...
 					default:
-						// STRING and BYTES types can have optional length
+						// STRING and DECIMAL types can have optional length
 						// suffixes, so only examine the prefix of the type.
 						if strings.HasPrefix(coltypes[cols[si]], "STRING") {
 							ivals[si] = parser.NewDString(string(t)).String()
-						} else if strings.HasPrefix(coltypes[cols[si]], "BYTES") {
-							ivals[si] = parser.NewDBytes(parser.DBytes(t)).String()
+						} else if strings.HasPrefix(coltypes[cols[si]], "DECIMAL") {
+							ivals[si] = fmt.Sprintf("%s", t)
```
Comment (tamird) [inline cli/dump.go]: why not string(t)?

File: cli/dump.go [hunk 2]
```diff
@@ -192,15 +192,15 @@ func dumpTable(w io.Writer, conn *sqlConn, origDBName, origTableName string) err
 					switch ct := coltypes[cols[si]]; ct {
 					case "INTERVAL":
 						ivals[si] = fmt.Sprintf("'%s'", t)
-					case "DECIMAL":
-						ivals[si] = fmt.Sprintf("%s", t)
+					case "BYTES":
+						ivals[si] = parser.NewDBytes(parser.DBytes(t)).String()
 					default:
-						// STRING and BYTES types can have optional length
+						// STRING and DECIMAL types can have optional length
```
Comment (tamird) [inline cli/dump.go]: BYTES can't?

**Comment by knz** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[cli/dump.go, line 198 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842#-KMefksuEPH2ZiAkHify-r1-198:-KMubq5AtBbIFYjkrZy_:-1202860624) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b70e3e97c9dd1d8f9d5461adab28ea92b601322/cli/dump.go#L198)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Correct. I thought they could, but both the docs and sql.y confirm BYTES can't.
</details>
Hmm my gut feeling is that this warrants a CockroachDB extension to optionally limit the size of BYTES. What do you think? (perhaps a different PR though)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842)*

**Comment by bdarnell** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[cli/dump.go, line 198 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842#-KMefksuEPH2ZiAkHify-r1-198:-KMzwLgIjxpTL-BT7z8C:1286184198) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b70e3e97c9dd1d8f9d5461adab28ea92b601322/cli/dump.go#L198)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Hmm my gut feeling is that this warrants a CockroachDB extension to optionally limit the size of BYTES. What do you think? (perhaps a different PR though)
</details>
MySQL has a `VARBINARY(n)` type, but in postgres only the character string types have this limit. In my experience the limits on `VARCHAR(n)` have been more of a nuisance than a help (a relic of the time when saving bytes on the length field of a string was worth thinking about), so I don't think it's worth introducing new syntax to set a limit here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[cli/dump.go, line 198 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842#-KMefksuEPH2ZiAkHify-r1-198:-KN-Yb7u2isQ2KYltbxO:227998854) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b70e3e97c9dd1d8f9d5461adab28ea92b601322/cli/dump.go#L198)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> MySQL has a `VARBINARY(n)` type, but in postgres only the character string types have this limit. In my experience the limits on `VARCHAR(n)` have been more of a nuisance than a help (a relic of the time when saving bytes on the length field of a string was worth thinking about), so I don't think it's worth introducing new syntax to set a limit here.
</details>
Limiting the length of a field does affect the column family selection heuristics, though I suppose you can manually specify the column family for a `BYTES` column if desired. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842)*

**Comment by tamird** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7842)*

**Pull request merged**